dataprofiler/tests/data_readers/test_data.py

```diff
@@ -160,6 +161,23 @@ def test_read_url_verify_ssl(self, mock_request_get):
         ):
             data_obj = Data("https://test.com")
 
+    @patch("boto3.client")
+    @patch(
+        "os.environ",
+        {
+            "AWS_ACCESS_KEY_ID": "<YOUR_ACCESS_KEY>",
+            "AWS_SECRET_ACCESS_KEY": "<YOUR_SECRET_KEY>",
+        },
+    )
+    def test_read_s3_uri(self, mock_boto3_client):
+        # Create a custom mock response
+        custom_response = {"Body": MagicMock()}
+        custom_response["Body"].read.return_value = b"Test S3 content"
+
+        # Mock the behavior of the S3 client to return the custom response
+        mock_boto3_client.return_value.get_object.return_value = custom_response
+        data_obj = Data("s3a://my-bucket/my_file.txt")
+
```

Comment:
I'd recommend a second new function here (`test_read_s3_uri_failure_logger_check`) too to handle [these](https://github.com/capitalone/DataProfiler/pull/1054/files#diff-6199c8b82b73b32f5fcaac940ca3bc280409012e690f09b354d4d86a1744b137R878-R879)  lines of code. 

You can use [`self.assertLogs`](https://docs.python.org/3.4/library/unittest.html#unittest.TestCase.assertLogs) to validate `f"'{path}' is not a valid S3 URI"` is logged

Reply:
Done. To do so, I needed to refactor `is_s3_uri` by adding a logger to ensure successful testing with `test_read_s3_uri_failure_logger_check` and verify the logging behavior in unit tests.